include/quic/endpoint.hpp

```diff
@@ -17,56 +17,118 @@ extern "C"
 #include <queue>
 #include <random>
 #include <string>
-#include <unordered_map>
 #include <uvw.hpp>
 
 #include "connection.hpp"
 #include "context.hpp"
+#include "network.hpp"
 #include "utils.hpp"
 
 namespace oxen::quic
 {
-    class Connection;
-    class Handler;
-
-    class Endpoint
+    class Endpoint : std::enable_shared_from_this<Endpoint>
     {
+        friend class Network;
         friend class Connection;
+        friend class Stream;
 
-      public:
-        explicit Endpoint(std::shared_ptr<Handler>& quic_manager);
-        virtual ~Endpoint() { log::trace(log_cat, "{} called", __PRETTY_FUNCTION__); }
+      private:
+        const Address local;
+        std::shared_ptr<uvw::timer_handle> expiry_timer;
+        std::shared_ptr<uv_udp_t> handle;
+        bool accepting_inbound{false};
+        Network& net;
 
-        std::shared_ptr<Handler> handler;
+      public:
+        explicit Endpoint(Network& n, const Address& listen_addr);
+
+        template <typename... Opt>
+        bool listen(Opt&&... opts)
```

Comment:
This function can only ever return true, so shouldn't be a `bool` return.

Reply:
Do we want to have a bool return to check if the server was initialized correctly? If so, I can have the promise set false and return the exception to console.